src/tests/unit/common/file-url-provider.test.tsx

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { Mock, Times } from 'typemoq';
+import { FileURLProvider } from '../../../common/file-url-provider';
+import { WindowUtils } from '../../../common/window-utils';
+
+describe('FileURLProviderTest', () => {
+    it('provideURL', () => {
+        const windowUtilsMock = Mock.ofType(WindowUtils);
+        const provideBlobMock = Mock.ofType<(blobParts?: any[], mimeType?: string) => Blob>();
+        const content = ['<a></a>'];
+        const mimeType = 'text/html';
+        const returnedURL = 'returned url';
+        const blobStub = {} as Blob;
+
+        provideBlobMock
+            .setup(pbm => pbm(content, mimeType))
+            .returns(() => blobStub)
+            .verifiable(Times.once());
+
+        windowUtilsMock
+            .setup(w => w.createObjectURL(blobStub))
```

Comment:
nit: You can rename `w` inside the `setup` function to a better name. `windowsUtilProvider` maybe?